pkgs/development/tools/symfony-cli/default.nix

```
@@ -12,12 +12,20 @@ buildGoModule rec {
     sha256 = "sha256-GAsyI8I+tHFMV/LqwPx2ph+w3zaqKSn9vieVQcuO+y0=";
   };
 
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
```

Are not these two different flags?

```suggestion
    "-X" "main.version=${version}"
```

---

I don’t really know if it should be considered as two different flags. If we wanted to add something like `main.channel` it would result in :  

```
ldflags = [
  "-s"
  "-w"
  "-X"
  "main.version=${version}"
  "-X"
  "main.channel=${channel}"
]
```

instead of :  


```
ldflags = [
  "-s"
  "-w"
  "-X main.version=${version}"
  "-X main.channel=${channel}"
]
```

I really don’t know what is the best for NixOS

---

The question is whether it works if you run it in shell.

The `stdenv.mkDerivation` does not really support lists by default so currently, it will end up being run like `go install "-ldflags=-s -w -X main.version=x.y.z -X main.channel=foo"`:

https://github.com/NixOS/nixpkgs/blob/fbcb61bd7eb19914cbd88789c3586a63ff46b72b/pkgs/build-support/go/module.nix#L211

But in the future, when we switch to structured attributes, it would be good if arguments would be properly split.